### PR TITLE
Sync workflow with latest docs

### PR DESCRIPTION
**Pull request by trask**
I've moved the documentation to https://github.com/trask/repository-template (hoping still to find an official place in OTel for it, waiting to see if it can be successfully adopted by at least one other language: https://github.com/open-telemetry/opentelemetry-python/pull/2649

Feel free to review/merge now, or wait until I run through the (approx) same changes in the upcoming instrumentation release: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/5998

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -91,7 +94,7 @@ jobs:
 
       - name: Create GitHub release
         env:
-          GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (trask) [inline .github/workflows/release.yml:97]: @jack-berg I think this change fixes https://github.com/open-telemetry/opentelemetry-java/runs/6356816939?check_suite_focus=true

```
HTTP 404: Not Found (https://api.github.com/repos/open-telemetry/opentelemetry-java/releases)
```

**Comment by trask** [comment]
TODO @trask fix "checks" box not closing

**Review by jack-berg — APPROVED** [review]

**Review by kubawach — APPROVED** [review]
LGTM :)

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by anuraaga** [comment]
@trask Is https://github.com/open-telemetry/opentelemetry-java/pull/4454#issuecomment-1121942320 fixed or will be fixed in this PR?

**Comment by trask** [comment]
> @trask Is [#4454 (comment)](https://github.com/open-telemetry/opentelemetry-java/pull/4454#issuecomment-1121942320) fixed or will be fixed in this PR?

yes, now 😄

**Review by anuraaga — APPROVED** [review]

**Pull request merged**